From da8f5d0260f1a4274ddad4266bddc3839e77401e Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Wed, 2 Dec 2020 12:11:33 +0100
Subject: [PATCH] Revert "ipq40xx: enable RX hash / CTAG TX offloading for
 single-phy"

This reverts commit e73d866c6fc7dfad2fb7dd87345c90029bbda59d.
---
 .../drivers/net/ethernet/qualcomm/essedma/edma_axi.c     | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/target/linux/ipq40xx/files/drivers/net/ethernet/qualcomm/essedma/edma_axi.c b/target/linux/ipq40xx/files/drivers/net/ethernet/qualcomm/essedma/edma_axi.c
index 49ee31a1bd..50335b0d14 100644
--- a/target/linux/ipq40xx/files/drivers/net/ethernet/qualcomm/essedma/edma_axi.c
+++ b/target/linux/ipq40xx/files/drivers/net/ethernet/qualcomm/essedma/edma_axi.c
@@ -978,20 +978,11 @@ static int edma_axi_probe(struct platform_device *pdev)
 		edma_netdev[i]->wanted_features = NETIF_F_HW_CSUM | NETIF_F_SG |
 					     NETIF_F_TSO | NETIF_F_GRO;
 
-		if (of_property_read_bool(np, "qcom,single-phy") && edma_cinfo->num_gmac == 1)
-			edma_netdev[i]->features |= NETIF_F_HW_VLAN_CTAG_TX;
-
 #ifdef CONFIG_RFS_ACCEL
 		edma_netdev[i]->features |=  NETIF_F_NTUPLE;
 		edma_netdev[i]->hw_features |=  NETIF_F_NTUPLE;
 		edma_netdev[i]->vlan_features |= NETIF_F_NTUPLE;
 		edma_netdev[i]->wanted_features |= NETIF_F_NTUPLE;
-		if (of_property_read_bool(np, "qcom,single-phy") && edma_cinfo->num_gmac == 1) {
-			edma_netdev[i]->features |= NETIF_F_RXHASH;
-			edma_netdev[i]->hw_features |= NETIF_F_RXHASH;
-			edma_netdev[i]->vlan_features |= NETIF_F_RXHASH;
-			edma_netdev[i]->wanted_features |= NETIF_F_RXHASH;
-		}
 #endif
 		edma_set_ethtool_ops(edma_netdev[i]);
 
-- 
2.25.1

